### PR TITLE
Add -u to go get

### PR DESCRIPTION
**Pull request by justinj**
---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/examples-orms/21)

### CODE REVIEW

**Comment by tamird** [comment]
This shouldn't be necessary, given the `go get -d -t ./...`

Consider running `go get -d -t -v ./...` locally if you're trying to figure out why that doesn't install testserver.

**Comment by justinj** [comment]
I think the problem was actually if you have an outdated version of testserver the existing command didn't update it, so we should just need to add `-u`, but that seems to have problems too that I'm looking into now.

File: Makefile [hunk 1]
```diff
@@ -17,8 +17,12 @@
 
 GO ?= go
 POSTGRES_TEST_TAG ?= 20170308-1644
-EXAMPLES_ORMS_PATH = /examples-orms
-DOCKER = docker run --volume="$(shell pwd)":$(EXAMPLES_ORMS_PATH) docker.io/cockroachdb/postgres-test:$(POSTGRES_TEST_TAG)
+EXAMPLES_ORMS_PATH = /root/go
```
Comment (benesch) [inline Makefile]: I'd make this:

```make
DOCKER_GOPATH := /root/go
DOCKER_REPO_PATH := $(DOCKER_GOPATH)/src/github.com/cockroachdb/examples-orms
```

and then use `DOCKER_GOPATH` and `DOCKER_REPO_PATH` appropriately below.

**Comment by tamird** [comment]
What is the problem you're fixing here? In general, `go get -u` won't do what you want if you're actively developing because `go get` will complain about a detached head.

As for the second bit where you mount the repo at the correct location in the container - why is that necessary? You're still not sharing the dependencies' sources with the host, so what's the gain?

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/examples-orms/21#-:-KkHu8wfRyZVqcmQgBd-:b-lseld)*

**Comment by justinj** [comment]
The problem is that if you have an outdated version of `cockroach-go` the the test will just hang, so this ensures that it gets updated when you run `make deps`. I'll add that to the commit message. Is running `make deps` in this repo a common occurrence when actively developing in it? I suspect it's just something you'd run in order to run all the tests against a new version of Cockroach, but I might be wrong.

The mounting is necessary because otherwise the mounted repo doesn't exist within a $GOPATH and `go get -u` fails (with a not-especially-clear error message).

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/examples-orms/21#-:-KkHudyUnFSjNDbM2ENO:b-qhvagu)*

**Comment by tamird** [comment]
Oh, I misread; you _are_ sharing the GOPATH with the host now; we should definitely not do that, because you'll pollute the build products produced on the host with those produced in the container.

>  Is running make deps in this repo a common occurrence when actively developing in it? I suspect it's just something you'd run in order to run all the tests against a new version of Cockroach, but I might be wrong.

I'm not the right person to answer that =/

> The mounting is necessary because otherwise the mounted repo doesn't exist within a $GOPATH and go get -u fails (with a not-especially-clear error message).

Got it.

As I said above: you probably only want to share GOPATH/src between host and container, not the entirety of GOPATH.

---

Reviewed 1 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Makefile, line 22 at r2](https://reviewable.io:443/reviews/cockroachdb/examples-orms/21#-KkHvQK92dNr7LoH-tRV:-KkHvQK92dNr7LoH-tRW:bqppqpp) ([raw file](https://github.com/cockroachdb/examples-orms/blob/9ae4f1a027d52c9f0d224d6a6603761fd14c8a57/Makefile#L22)):*
> ```Makefile
> DOCKER_GOPATH = /root/go
> DOCKER_REPO_PATH = $(DOCKER_GOPATH)/src/github.com/cockroachdb/examples-orms
> DOCKER = docker run --volume="$(shell pwd)/../../../..":$(DOCKER_GOPATH) docker.io/cockroachdb/postgres-test:$(POSTGRES_TEST_TAG)
> ```

You can also replace this dance with `${GOPATH%%:*}` to get the first entry in the GOPATH without relative directories and so forth.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/examples-orms/21#-:-KkHvIIPE8rD7OOPW9V-:b-wg9lsm)*

**Comment by benesch** [comment]
In general you don't need to update testserver, because testserver will download the latest bleeding edge binary from `edge-binaries.cockroachdb.com`. Somewhere around 1.0, though, we landed a change to move some log output from stdout to stderr, which broke the testserver's sniffing of the SQL url—causing the hang-forever that you saw, @justinj. So just this once you needed a new version of testserver. 1.0 stability guarantees should prevent breakage in the future.

Anyway, `go get -u` will fail whenever you're on a branch without an upstream configured. I'm fine with that. I don't see another good way to get updated copies of the dependencies, and I'm not sure we want to open the vendoring can of worms.

**Comment by benesch** [comment]
Looks splendid to me. @tamird PTAL.

**Comment by tamird** [comment]
:lgtm: mod slight inaccuracy in commit message (we don't mount the entire GOPATH).

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/examples-orms/21#-:-KkI1tuNIBk97lk4ACMn:b-lqvmbs)*

**Comment by justinj** [comment]
Merging despite the extant failure unrelated to this change.

**Pull request merged**